homeassistant/components/bluesound/button.py

```diff
@@ -0,0 +1,129 @@
+"""Button entities for Bluesound."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from pyblu import Player
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.const import CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import (
+    CONNECTION_NETWORK_MAC,
+    DeviceInfo,
+    format_mac,
+)
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import BluesoundCoordinator
+from .media_player import DEFAULT_PORT
+from .utils import format_unique_id
+
+if TYPE_CHECKING:
+    from . import BluesoundConfigEntry
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: BluesoundConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Bluesound entry."""
+
+    buttons = [
+        BluesoundButton(
+            config_entry.runtime_data.coordinator,
+            config_entry.runtime_data.player,
+            config_entry.data[CONF_PORT],
+            description,
+        )
+        for description in BUTTON_DESCRIPTIONS
+    ]
+
+    async_add_entities(buttons)
+
+
+@dataclass(kw_only=True, frozen=True)
+class BluesoundButtonEntityDescription(ButtonEntityDescription):
+    """Description for Bluesound button entities."""
+
+    press_fn: Callable[[Player], Awaitable[None]]
+
+
+async def clear_sleep_timer(player: Player) -> None:
+    """Clear the sleep timer."""
+    sleep = -1
+    while sleep != 0:
+        sleep = await player.sleep_timer()
+
+
+async def set_sleep_timer(player: Player) -> None:
+    """Set the sleep timer."""
+    await player.sleep_timer()
+
+
+BUTTON_DESCRIPTIONS = [
+    BluesoundButtonEntityDescription(
+        key="set_sleep_timer",
+        name="Set Sleep Timer",
+        press_fn=set_sleep_timer,
+    ),
+    BluesoundButtonEntityDescription(
+        key="clear_sleep_timer",
+        name="Clear Sleep Timer",
```

Comment:
Let's use entity translations for this. Assign a translation_key and make a reference in strings.json.

(You can also use that translation key to add a nice icon in icons.json)